### PR TITLE
Redux

### PR DESCRIPTION
**Pull request by muhammadali-rakhimov**

### CODE REVIEW

**Comment by muhammadali-rakhimov** [comment]
Redux added and fetching

File: src/pages/App.js [hunk 1]
```diff
@@ -1,16 +1,72 @@
...
+  settingMovies,
+  settingTotalAmount,
+  sortBy,
+  sortOrder,
+  filter,
+  limit,
+  search,
+  setClearSearchMovies,
+}) {
+  const { data, isSuccess, isError, error, isLoading } = useGetMoviesQuery({
```
Comment (elkinuk) [inline src/pages/App.js:49]: With Redux all side-effects should be done inside of the Redux. That's why we need saga or thunk libs: to make async actions possible in Redux

Everything you did here with data should be moved in some Redux action like getMoviesData (result of this action will be = movies data is fetched, settingMovies or settingTotalAmount should not be exposed to React at all - it shouldn't know about existance of something where data is set per se), in app you'd be dispatching it and all requesting and setting data in the store will happen under the Redux hood. React doesn't orchestrate how data is handled, written or read. It just calls simplest action possible

**Review by elkinuk — CHANGES_REQUESTED** [review]

File: src/pages/App.js (hunk 1 above)
Comment (muhammadali-rakhimov) [reply src/pages/App.js:49]: So you want me to change to redux thunk | saga from rtk query completely. Wow, there is so many things that I must change. implementing, testing... 😮😐

Comment (muhammadali-rakhimov) [reply src/pages/App.js:49]: image

**Pull request merged**